apps/gitness/src/pages/pull-request/pull-request-compare-page.tsx

```diff
@@ -141,6 +141,14 @@ export const CreatePullRequest = () => {
     memorizedState
   ])
   const { data: { body: repoMetadata } = {} } = useFindRepositoryQuery({ repo_ref: repoRef })
+
+  useEffect(() => {
+    if ((!diffTargetBranch || !diffSourceBranch) && repoMetadata?.default_branch) {
+      setSelectedTargetBranch(diffTargetBranch ?? repoMetadata?.default_branch)
+      setSelectedSourceBranch(diffSourceBranch ?? repoMetadata?.default_branch)
+    }
+  }, [repoMetadata])
```

Comment:
do we need to add `diffTargetBranch` and `diffSourceBranch` to effect deps?

Reply:
no, they are either always present in params when you move to a PR creation page via new commit, or never present if you open a create new PR page